### DISCUSSION TITLE
Java Question

### DISCUSSION THREAD

**Discussion by EthanKC** [nextgenhealthcare/connect#6008] [Q&A]
I've been looking into which versions of Java you can use with Mirth. From what I can tell, the answer is currently anything 11+. Is that always true? When a new version of Java releases, is Mirth already able to use it? Or would there need to be a new version of Mirth that is compatible? 

I'm asking because I'm going to be installing several instances of Mirth on different VMs. The version of Java on these VMs isn't something I support, so I want to know, if the version of Java gets upgraded, do I need to be concerned that I'll have instances of Mirth spread across different VMs that suddenly stop working?

**Comment by pacmano1** [comment]
Sorta impossible to know if some future release of java is guaranteed to be compatible, right?   

So your devops people should not upgrade underlying prerequisites for software used in your enviroments without specific approval from the engineering team.   Sounds like poor practice otherwise.

**Comment by odoodo** [comment]
Actually it's anything 8+.

**Reply by pacmano1** [reply]
Yea, but given newer versions of java needed extra vmoptions it's obvious there are no guarantees of compatibility.
